containers/Comments/CommentForm.js

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { Formik } from 'formik';
+import * as yup from 'yup';
+import { TextInputFormField } from '../../components/formFields';
+import { ButtonSubmit } from '../../components/ButtonSubmit';
+
+export const CommentForm = (props) => {
+
+  const commentValidationSchema = yup.object().shape({
+    comment: yup
+      .string()
+      .max(4000, 'Comment must be no longer than 4000 characters.')
+      .required('Please enter you comment.'),
+  });
+
+  return (
+    <Formik
+      validationSchema={commentValidationSchema}
+      initialValues={{
+        comment: '',
+      }}
+      onSubmit={(values, {resetForm}) => {props.handleSubmit(values); resetForm({comment: ''});}}
+    >
+      {({
+        handleChange,
+        handleBlur,
+        handleSubmit,
+        values,
+        errors,
+        touched,
+        isValid,
+      }) => (
+        <>
+          <TextInputFormField
+            name='comment'
+            value={values.comment}
+            placeholder='Enter your comment'
+            onChangeText={handleChange('comment')}
+            onBlur={handleBlur('comment')}
+            error={errors.comment}
+            touched={touched.comment}
+            autoFocus = {props.isFocused}
```

Comment:
@khamui This is what doesnt work, so the form does not get focused when pops.isFocused is set to true. I also tried with extra state in CommentForm, that also did not help.

Reply:
i pushed something very drafty (i tried to make it nice, but it was not as easy as i thought in the beginning though).

What could be the problem that `isFocused` does not work here is that `CommentForm` has been rendered already and while it is still `living` the change of this focus variable does not have an effect on it.

In my commit I make it dependable on a variable called `showCommentForm` which controls the entire component to be rendered or not. This way a change of that variable causes a new render of the component and therefore the variable applies.

Note: I am not sure if this behavior is what we want or not. Maybe we can have a quick talk tomorrow? Maybe @phillimorland wants to join as well?

Reply:
@khamui Cool, the focusing works! Although I discussed the functionality a bit different with luca: the comment form should always be there at the bottom (without having to click on add comment) and when the user clicks on reply it gets focused (thats why I named it commentFormFocused).. but we can also discuss it again with him?
With the current version (probably also before), there is a problem when you clicked on add comment, then dont add one and click on reply -> nothing happens ?

Reply:
i see. actually my drafty version is just an attempt and i agree on the user flow it is a bit weird.
Currently the focus only works if the CommentForm is re-rendered, either clicking on the SubmitButton or the ReplyButton. A `"trigger only by focus"` solution we would need to research and tryout a little bit i guess.

Yes lets have a chat about!